packages/qwik/src/core/shared/component-execution.ts

```diff
@@ -101,7 +102,9 @@ export const executeComponent = (
       (jsx) => {
         const useOnEvents = container.getHostProp<UseOnMap>(renderHost, USE_ON_LOCAL);
         if (useOnEvents) {
-          return maybeThen(addUseOnEvents(jsx, useOnEvents), () => jsx);
+          return maybeThen(addUseOnEvents(jsx, useOnEvents), (_jsx) => {
+            return _jsx;
```

Comment:
Actually, the maybeThen is only here to catch errors, and if it errors it will now return the error object. 

@Varixo can this even error?

Reply:
we use this because we need to await for addUseOnEvents if it is a promise

Reply:
@JerryWu1234 Wout is right. Can we just return `addUseOnEvents(jsx, useOnEvents)` directly?

Reply:
done